scripts/write-api-docs.js

```diff
@@ -0,0 +1,939 @@
+        if (! call || call.length === 0) {
+          throw null
+        }
+        call[0].arguments.forEach(arg => {
+          assertType(arg, OBJECT_TYPES, filePath)
+          merge(result, marshallParameterProperties(docs, paramTag, routeSlug, arg, variables))
+        })
+        break
+      }
+      default:
+        throw null
+      /* eslint-enable indent */
+    }
+
+    return result
+  }, [])
+
+  function merge (target, source) {
+    source.forEach(item => {
+      const paramName = item.paramName
+      if (invertedIndex.has(paramName)) {
+        const resultIndex = invertedIndex.get(paramName)
+        if (target[resultIndex].validation !== item.validation) {
+          target[resultIndex].validation += `;<br />or ${item.validation}`
+        }
+      } else {
+        invertedIndex.set(paramName, target.length)
+        target.push(item)
+      }
+    })
+
+    return target
+  }
+}
+
+function marshallParameterProperties (docs, paramTag, routeSlug, parameters, variables) {
+  return parameters.properties.map(parameter => ({
+    paramName: parameter.key.name,
+    paramDescription: docs[camel(`${paramTag}-${routeSlug}-${parameter.key.name}`)],
+    validation: `*${marshallValidation(marshallValue(parameter.value, variables))}*`
+  }))
+}
+
+function marshallRequestBodyParameters (docs, routeSlug, validation, variables, filePath) {
+  return marshallParameters(docs, 'request-body', routeSlug, validation, 'payload', variables, filePath)
+}
+
+function marshallResponseBodyParameters (docs, routeSlug, response, variables, filePath) {
+  return marshallParameters(docs, 'response-body', routeSlug, response, 'schema', variables, filePath)
+}
+
+function findRouteHandler (route, filePath) {
+  try {
+    // HACK: Assumes route handlers are defined inline in the object literal
+    return findProperty(route, 'handler', FUNCTION_EXPRESSION_TYPES, filePath)
+  } catch (error) {
+  }
+}
+
+function marshallErrorResponses (errors, handler, filePath) {
+  const dupes = new Set()
+  // HACK: Assumes we always import error module as `error`
+  return find(handler, {
+    type: 'CallExpression',
+    callee: {
+      type: 'MemberExpression',
+      object: {
+        type: 'Identifier',
+        name: 'error'
+      }
+    }
+  }, { recursive: true })
+  .filter(errorCall => {
+    const errorName = errorCall.callee.property.name
+    if (dupes.has(errorName)) {
+      return false
+    }
+    dupes.add(errorName)
+    return !! errors[errorName]
+  })
+  .map(errorCall => errors[errorCall.callee.property.name])
+}
+
+function marshallAuthentication (authentication) {
+  if (! authentication) {
+    return
+  }
+
+  const tokens = authentication.tokens.map(token => {
+    return marshallToken(token)
+  }).reduce((deduped, token) => {
+    if (deduped.indexOf(token) === -1) {
+      deduped.push(token)
+    }
+    return deduped
+  }, [])
+
+  return {
+    emojis: `:lock:${authentication.optional ? ':question:' : ''}`,
```

Comment:
let's also put ( ) back (if it is easy enough) like it was 
![image](https://cloud.githubusercontent.com/assets/128755/25187666/b8c601f8-24f1-11e7-8cd6-b3ebfce0051c.png) , new docs are missing those